src/postamble.js

```diff
@@ -194,21 +194,7 @@ function callMain(args) {
     exit(ret, /* implicit = */ true);
   }
   catch (e) {
-    // Certain exception types we do not treat as errors since they are used for
-    // internal control flow.
-    // 1. ExitStatus, which is thrown by exit()
-    // 2. "unwind", which is thrown by emscripten_unwind_to_js_event_loop() and others
-    //    that wish to return to JS event loop.
-    if (e instanceof ExitStatus || e == 'unwind') {
-      return;
-    }
-    // Anything else is an unexpected exception and we treat it as hard error.
-    var toLog = e;
-    if (e && typeof e === 'object' && e.stack) {
-      toLog = [e, e.stack];
-    }
-    err('exception thrown: ' + toLog);
-    quit_(1, e);
+    handleException(e);
```

Comment:
quit_ is different than what `handleException` does, I think. e.g. on node it calls `process['exit'](status)`.

Reply:
True, but I think the fact that we don't see any test failure here points to the fact that this is safe change.

Basically it means that rather then quitting the process directly here when main causes an exception we let the exception propogate out where it will become either an uncaught exception or and uncaught promise rejection.  Both of these then get handled by shell.js by default (because both NODEJS_CATCH_EXIT and NODEJS_CATCH_REJECTION are both enabled by default).  

For the web use case there is no difference because all `quit_`  does is re-throw the exception.

So this is a node-only issue, and I believe we have good coverage of the exit process these days.

Reply:
Also, I think is much more polite to the surrounding/embedding node code to let exceptions propagate there rather than bringing down the entire not process (this is never desirable in a node.js context for example).

Reply:
WDYT?

Reply:
This sounds reasonable, but if so, do we still need `quit_`? The only other usage is in `procExit()`, and it seems like the same arguments apply there, but I may be missing something.

Reply:
Thats a good point.... I think we should do that too.   Do you think I should extent this PR to remove `quit_` too?  Or make that followup?

Reply:
I think it can be in this PR. But let's make sure it's the right thing. My concern is that `quit_` sort of sounds right when running an entire program - say a commandline tool that was compiled into a JS+wasm combination. It should quit like a node application would, and not throw an error. But, for a library, everything you said makes sense. However I am not sure we can differentiate those use cases, even if what I wrote here makes sense.

Reply:
I decided to keep the quit_ function call it from here.